src/CmakeRegen.cpp

```diff
@@ -186,30 +186,31 @@ void RegenerateCmakeAddSubdirectory(std::ostream& o,
     for (auto subdir : subdirs) {
         o << "add_subdirectory(" << subdir << ")\n";
     }
-        
+
 }
 
 void RegenerateCmakeAddTarget(std::ostream& o,
                               const Configuration& config,
-                              const Component& comp,
+                              Component& comp,
                               const std::list<std::string>& files,
                               bool isHeaderOnly) {
     o << comp.type << "(${PROJECT_NAME}";
 
     if (isHeaderOnly) {
-        o << " INTERFACE\n";
-    } else {
-        if (config.addLibraryAliases.count(comp.type) == 1) {
-            o << " STATIC\n";
-        } else {
-            o << "\n";
-        }
+        comp.additionalTargetParameters.insert("INTERFACE");
```

Comment:
This is definitely the wrong place to do this. This function really shouldn't change this argument.

Reply:
Totally agree, was the easiest way to achieve this ;). I will lift it to the input side to have there header only detection.